scripts/nf-core

```diff
@@ -57,6 +57,32 @@ def list(json):
     """ List nf-core pipelines with local info """
     nf_core.list.list_workflows(json)
 
+@nf_core_cli.command()
+@click.argument(
+    'pipeline',
+    required = True,
+    metavar = "<pipeline name>"
+)
+@click.option(
+    '-r', '--release',
+    type = str,
+    help = "Pipeline release"
+)
+@click.option(
+    '-s/-n', '--singularity/--no-singularity',
```

Comment:
Wouldn't it be better to have something like `-s yes` / `-s no` instead?
I find it disturbing to have two options for two opposites things.

Reply:
I guess - the above is the standard way to handle boolean flags with click (see [docs](http://click.pocoo.org/5/options/#boolean-flags)), so this is the way I've done it in the past. But I don't really mind either way.

Reply:
OK, I didn't knew the standards.
I'm fine with it then ;-)

Reply:
Note that I did it as a single flag first - just `-s` to get singularity images and nothing to not get them. But then I thought it was better to be explicit and require this input from the user.

Reply:
Yeah, after playing a bit more I agree. As it's required the help message says that it's missing `-s` if not supplied, which is kind of counter-intuitive.

Just need to find a nice way to handle `y|Y|yes|Yes|YES` for both options now. I bet `click` has a nice tidy way to handle this for you, if I can just find it..

Reply:
I too would prefer just one parameter `-s` or `--singularity-images` with yes/no as only accepted values, that can be done by using [choices](http://click.pocoo.org/5/options/#choice-options) or giving a `call_back`. 

Or just `--singularity/--no-singularity` (without `-s/-n`)

Reply:
Yes - I half implemented the `Choice` thing this morning, but if we want to handle case insensitivity and short/long form then the list of allowed options gets messy... 🤓 (`y|Y|yes|Yes|YES|n|N|no|No|NO`).

We could write a custom callback or string handling behaviour, but that feels like overkill and the help text is not particularly pretty then. `click` already has this kind of handling for the [`--yes` behaviour](http://click.pocoo.org/5/options/#yes-parameters). `click.BOOL` can [also handle string values](http://click.pocoo.org/5/parameters/#parameter-types), so I figure that click probably has built-in support for this _somehow_...

Reply:
Yeah!! looks like setting `type=click.BOOL` does the trick right ? for *y/n* values

Option:
```
@click.option('-s', '--singularity', type=click.BOOL, default=None, required=True)
```

Test:
```
$ python cli_test.py test -s y
True
$ python cli_test.py test -s yes
True
$ python cli_test.py test -s YEs
True
$ python cli_test.py test -s YeS
True
$ python cli_test.py test -s nO
False
$ python cli_test.py test -s n
False
$ python cli_test.py test -s NO
False
$ python cli_test.py test -s NOo
Usage: cli_test.py test [OPTIONS]

Error: Invalid value for "-s" / "--singularity": noo is not a valid boolean
$ python cli_test.py test -s 0
False
$ python cli_test.py test -s 1
True
 $ python cli_test.py test -s 2
Usage: cli_test.py test [OPTIONS]

Error: Invalid value for "-s" / "--singularity": 2 is not a valid boolean
```